lib/util/package-rules/index.ts

```diff
@@ -73,25 +73,27 @@ export function applyPackageRules<T extends PackageRuleInputConfig>(
     // This rule is considered matched if there was at least one positive match and no negative matches
     if (matchesRule(config, packageRule)) {
       // Package rule config overrides any existing config
-      const toApply = { ...packageRule };
+      const toApply = removeMatchers({ ...packageRule });
       if (config.groupSlug && packageRule.groupName && !packageRule.groupSlug) {
         // Need to apply groupSlug otherwise the existing one will take precedence
         toApply.groupSlug = slugify(packageRule.groupName, {
           lower: true,
         });
       }
       config = mergeChildConfig(config, toApply);
-      delete config.matchPackageNames;
-      delete config.matchPackagePatterns;
-      delete config.matchPackagePrefixes;
-      delete config.excludePackageNames;
-      delete config.excludePackagePatterns;
-      delete config.excludePackagePrefixes;
-      delete config.matchDepTypes;
-      delete config.matchCurrentValue;
-      delete config.matchCurrentVersion;
-      delete config.matchCurrentAge;
     }
   }
   return config;
 }
+
+function removeMatchers(
+  packageRule: PackageRule & PackageRuleInputConfig,
+): Record<string, unknown> {
+  for (const key of Object.keys(packageRule)) {
+    if (key.startsWith('match') || key.startsWith('exclude')) {
```

Comment:
we should add a test which validates that every option starting with those prefixes are only allowed in package rules. so we don't silently delete wrong keys in future.

Reply:
Its already caught during config validation, where we check that the children is inside the correct parent or not.
https://github.com/renovatebot/renovate/blob/f69d7e83b33727cb66742facafb24e1c3ce1e432/lib/config/validation.ts#L235-L237

Reply:
that's not what I like to test.

I like to ensure nobody adds a `matchSomething` option to any other parent than package rules.

Reply:
Well, they might be added to something like hostRules but then it's also safe to assume we can delete them from packageRules if found as they shouldn't be there in the first place and hopefully found during config validation

Reply:
ok, here is a concrete sample
- https://github.com/renovatebot/renovate/pull/26951/files#diff-0053f214ec552bf58e21e52b0af7fcfe1699c7be2989cee50c52d44c7046ed2fR2831

but you're right, this should not be here in that case